metadata-ingestion/src/datahub/ingestion/source/dbt.py

```diff
@@ -98,17 +98,20 @@ def extract_dbt_entities(
         dbtNode = DBTNode()
 
         dbtNode.dbt_name = key
-        dbtNode.node_type = node["resource_type"]
-        dbtNode.name = node["name"]
         dbtNode.database = node["database"]
         dbtNode.schema = node["schema"]
         dbtNode.dbt_file_path = node["original_file_path"]
+        dbtNode.node_type = node["resource_type"]
+        if "identifier" in node and load_catalog is False:
```

Comment:
Minor concern, there's no tests on this component and the integration test won't cover this.  

IMO I'd prefer to add tests before adding conditional logic but I think this is fine as there's no unit tests ATM.